src/world/FieldMessage.cpp

```diff
@@ -0,0 +1,215 @@
+      half_line = newLine;
+    } else if (newLine.name == "center_line") {
+      center_line = newLine;
+    } else if (newLine.name == "left_penalty_line") {
+      left_penalty_line = newLine;
+    } else if (newLine.name == "right_penalty_line") {
+      right_penalty_line = newLine;
+    }
+      // adding rectangle box lines
+    else if (newLine.name == "top_left_penalty_stretch") {
+      top_left_penalty_stretch = newLine;
+    } else if (newLine.name == "bottom_left_penalty_stretch") {
+      bottom_left_penalty_stretch = newLine;
+    } else if (newLine.name == "top_right_penalty_stretch") {
+      top_right_penalty_stretch = newLine;
+    } else if (newLine.name == "bottom_right_penalty_stretch") {
+      bottom_right_penalty_stretch = newLine;
+    }
+}
+
+float FieldMessage::mm_to_m(float scalar) {
+  return scalar/1000;
+}
+
+Vector2 FieldMessage::mm_to_m(Vector2 vector) {
+  return {vector.x/1000, vector.y/1000};
+}
+
+// getters And setters
+double FieldMessage::field_width(){
+  return width;
+}
+double FieldMessage::field_length(){
+  return length;
+}
+double FieldMessage::goal_width(){
+  return goalWidth;
+}
+double FieldMessage::goal_depth(){
+  return goalDepth;
+}
+double FieldMessage::boundary_width(){
+  return boundaryWidth;
+}
+FieldLineSegment FieldMessage::getTop_line(){
+  return top_line;
+}
+FieldLineSegment FieldMessage::getBottom_line(){
+  return bottom_line;
+}
+FieldLineSegment FieldMessage::getLeft_line(){
+  return left_line;
+}
+FieldLineSegment FieldMessage::getRight_line(){
+  return right_line;
+}
+FieldLineSegment FieldMessage::getHalf_line(){
+  return half_line;
+}
+FieldLineSegment FieldMessage::getCenter_line(){
+  return center_line;
+}
+FieldLineSegment FieldMessage::getLeft_penalty_line(){
+  return left_penalty_line;
+}
+FieldLineSegment FieldMessage::getRight_penalty_line(){
+  return right_penalty_line;
+}
+FieldArc  FieldMessage::getTop_left_penalty_arc(){
+  return top_left_penalty_arc;
+}
+FieldArc  FieldMessage::getBottom_left_penalty_arc(){
+  return bottom_left_penalty_arc;
+}
+FieldArc  FieldMessage::getTop_right_penalty_arc(){
+  return top_right_penalty_arc;
+}
+FieldArc  FieldMessage::getBottom_right_penalty_arc(){
+  return bottom_right_penalty_arc;
+}
+FieldLineSegment FieldMessage::getTop_left_penalty_stretch(){
+  return top_left_penalty_stretch;
+}
+FieldLineSegment FieldMessage::getBottom_left_penalty_stretch(){
+  return bottom_left_penalty_stretch;
+}
+FieldLineSegment FieldMessage::getTop_right_penalty_stretch(){
+  return top_right_penalty_stretch;
+}
+FieldLineSegment FieldMessage::getBottom_right_penalty_stretch(){
+  return bottom_right_penalty_stretch;
+}
+FieldArc FieldMessage::getCenter_circle(){
+  return center_circle;
+}
+
+std::vector<FieldLineSegment> FieldMessage::getField_lines(){
+  return field_lines;
+}
+std::vector<FieldArc> FieldMessage::getField_arcs(){
+  return field_arcs;
+}
+
+void FieldMessage::invert() {
```

Comment:
You need to swap the left and right lines since they literally 'take eachother's place' when yous wap the field for example.

Reply:
Yeah I made an issue about this